### ISSUE TITLE
Better Documentations ?

### ISSUE THREAD

**Issue by yunstanford** [squeaky-pl/japronto#77]
I can't find any documentation, README is not enough for a web framework. readthedoc can help you in some sense.

**Comment by squeaky-pl** [comment]
There is somehow narrative documentation by example in: https://github.com/squeaky-pl/japronto/tree/master/examples

**Comment by yunstanford** [comment]
yeah, i'd like to have some documentations like http://aiohttp.readthedocs.io/en/stable/web.html, for example, request/response/server/client etc. that would be very helpful. With only the README, i don't know even how to use request object.

**Comment by squeaky-pl** [comment]
https://github.com/squeaky-pl/japronto/blob/master/tutorial/4_request.md

**Comment by yunstanford** [comment]
how can i get the query parameters from request obj ?

like, "http://0.0.0.0:8080/profile/?query_id=012345678"

**Comment by squeaky-pl** [comment]
This is exactly documented in the file I linked:

```# Given a HTTP 1.1 `GET` request to `/basic?a=1` this would yield
# `method` set to `GET`, `path` set to `/basic`, `version` set to `1.1`
# `query_string` set to `a=1` and `query` set to `{'a': '1'}`.```

**Comment by yunstanford** [comment]
doc in comment is not very noticeable. why not use some auto-doc system ? that'll be very helpful.

Thanks for your reply.

**Comment by squeaky-pl** [comment]
This project is intended for speed enthusiasts, people who like plumbing, exprimenting and early adopters. Whatever you see is a proof of concept. There is no point in writing nice documentation at this point since I know I am gonna break all the APIs anyway. That's why I don't want to encourage people to use it. If you are looking for somehow stable documented framework then aiohttp indeed or Sanic is for you.

**Comment by cpersico** [comment]
Hi, I've been playing with japronto and so far I'm having a great time experimenting. But I'm not yet clear on how to approach unit testing, I get it that it's still a proof of concept in many ways, but perhaps there's a recommended way of doing unit tests?
